analytics/management/commands/create_ga_report.py

```diff
@@ -0,0 +1,325 @@
+        output = []
+        output.append(page_data['title'])
+        output.append(page_data['url'])
+        published_date = page_data['published_date'].strftime('%Y-%m-%d')
+        output.append(published_date)
+        output.append(unicode(page_data['days_published']))
+        output.append(page_data['parent_page'])
+        output.append(page_data['category'])
+        types = []
+        if page_data['is_video']:
+            types.append('Video')
+        if page_data['is_interview']:
+            types.append('Interview')
+        if page_data['is_visualization']:
+            types.append('Visualization')
+        output.append(','.join(types))
+        word_count = sum([x for x in page_data['word_count_by_type'].values()])
+        output.append(unicode(word_count))
+        return output
+
+    def _format_ga_data_for_output(self, ga_statistics, other_statistics):
+        # TODO: Here is the perfect example of why this Command isn't generic enough...
+        output = [unicode(x) for x in ga_statistics]
+        json_data = {
+            'facebook': [0, 0],
+            'twitter': [0, 0],
+            'direct': [0, 0],
+            'google': [0, 0]
+        }
+        for key in json_data.keys():
+            filtered_stats = [x for x in other_statistics if key in x[0]]
+            for stats in filtered_stats:
+                json_data[key][0] += int(stats[1])
+                json_data[key][1] += int(stats[2])
+        output.extend(unicode(x) for x in json_data['facebook'])
+        output.extend(unicode(x) for x in json_data['twitter'])
+        output.extend(unicode(x) for x in json_data['direct'])
+        output.extend(unicode(x) for x in json_data['google'])
+        return output
+
+    def _calculate_days_published(self, published_date, end_date):
+        # TypeError: can't subtract offset-naive and offset-aware datetimes
+        date_diff = end_date - published_date.replace(tzinfo=None)
+        return date_diff.days
+
+    # TODO: Investigate the possibility of simply extending all of our models/snippets/etc. with a Mixin that provides
+    # a word_count function...
+    def _calculate_word_count_by_type(self, page):
+        types_to_ignore = [u'Embed', u'FullBleed', u'Image', u'Overflow', u'SectionBreak', u'Sharable']
+        word_count_stats = {}
+        # Generic 'body'...
+        if page.body:
+            for data in page.body.stream_data:
+                type = data['type']
+                if type in types_to_ignore:
+                    continue
+                if type not in word_count_stats:
+                    word_count_stats[type] = 0
+                value = data['value']
+                try:
+                    text = value['text']
+                except KeyError:
+                    word_count_stats[type] += 1
+                    continue
+                except TypeError:
+                    text = value
+                # 'List' type...
```

Comment:
Could be doing this more accurately with `isintance` or `if "name" in dict`? Catching exception is a big part of doing duck typing, but when those exceptions can be imprecise it could be a problem. KeyError is pretty precise but TypeError could come from a few different causes.